src/pages/index.js

```diff
@@ -15,39 +18,86 @@ const profile = document.querySelector(".profile");
 const profileEditButton = profile.querySelector(".profile__edit-button");
 const profileAddButton = document.querySelector(".profile__add-button");
 
+// const profile image edit buttton
+const profileImageButton = document.querySelector(".profile__image-edit");
+
 // all forms
 const forms = Array.from(document.forms);
 const formObjects = {};
 
+// hold card to delete
+let cardToDelete = null;
+
+const api = new API({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "3352b458-30c6-41be-9a42-00c52b81b385",
+    "Content-Type": "application/json",
+  },
+});
+
 // user info manipulation class
 const userInfo = new UserInfo({
   nameSelector: ".profile__name",
   jobSelector: ".profile__avocation",
+  photoSelector: ".profile__avatar",
 });
 
 // profile modal popup
 const profileModalPopup = new PopupWithForm({
   popupSelector: ".profile-modal",
   handleSubmitFunc: (data) => {
-    profileModalPopup.close();
-    userInfo.setUserInfo({
-      name: data.name,
-      job: data.job,
-    });
+    profileModalPopup.toggleSubmitText(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```